src/components/ebay-notice/index.js

```diff
@@ -16,30 +18,53 @@ const constants = {
 
 const defaults = {
     type: 'page',
-    status: 'confirmation'
+    status: 'priority',
+    hidden: false,
+    dismissible: false
 };
 
-function getTemplateData(state, input) {
+function getInitialState(input) {
+    const hidden = defaults.hidden;
     const type = input.type || defaults.type;
     const headingTag = (type === 'page' && input.headingLevel) ? `h${input.headingLevel}` : constants[type].headingTag;
     const status = input.status || defaults.status;
+    const dismissible = (input.dismissible && type === 'page') || defaults.dismissible;
 
     return {
         mainTag: constants[type].mainTag,
         headingTag: headingTag,
         contentTag: constants[type].contentTag,
+        dismissible,
         type,
         status,
+        hidden,
         ariaText: input.ariaText || '',
+        ariaLabelClose: input.ariaLabelClose || '',
         htmlAttributes: processHtmlAttributes(input),
-        renderBody: input.renderBody,
         mainClass: [`${type}-notice`, `${type}-notice--${status}`, input.class],
         headingClass: `${type}-notice__status`,
         contentClass: `${type}-notice__content`
     };
 }
+function getTemplateData(state) {
+    return state;
+}
+
+function init() {
+    observer.observeRoot(this, ['hidden'], (hidden) => {
+        emitAndFire(this, 'notice-change', { hidden });
+    });
+}
+
+function onDismiss() {
+    this.setState('hidden', true);
+    emitAndFire(this, 'notice-change', { hidden: true });
```

Comment:
If there is already an observer on this property (line 54 above), then why do we need to run emitAndFire again?

Reply:
This in case the dismiss button is explicitly clicked, and not changed through the hidden property

Reply:
But the click handler does change the hidden property, doesn't it?

Reply:
Yes it does, but the native setstate doesn't a callback so I am setting it explicitly.

Reply:
@yomed Is this correct? I thought the propertyObserver created in the init() method would catch this.

Reply:
Yes, this is right. The propertyObserver just observes the property on the DOM element, not the state value.  Using UI and API take different code paths initially, though they can share common processing afterwards. This particular example is super simple, so I'm not too concerned with the current setup.

Some examples of other patterns:
1) Marko has hooks to call a function whenever a certain value is updated. So we could create an `update_hidden()` function (the Marko hook) and then use that to call `emitAndFire()`.
2) We can create our own "common processing" function, which is roughly what we're doing now in Menu.

Reply:
Thanks for clarifying. Pattern 1 sounds good to me, but I guess we need to be consistent with menu.

Reply:
Also, this isn't a PR blocker, as Yoni says this case is super simple right now, but we should be very wary if we start doing this in multiple places or in more complex components than this.